status-im/status-react
# Fixes #4223: No error when scanning incorrect public key QR code

Fixes #4223: No error when scanning incorrect public key QR code

### Summary:

If an error occurs scanning the QR code when adding a contact, display the error message and return to the home screen. Screenshots attached.

### Steps to test:
- Open Status
- Test adding a contact with an invalid QR code. Two cases:
-- Completely invalid QR code, e.g. shop receipt, should see "Please enter a valid contact code"
-- User's own QR code: should see "You can't add yourself"
- Test can still add a contact with a valid QR code.

status: ready

![issue4223_screeniea](https://user-images.githubusercontent.com/19371062/40625208-89ef74fc-62a0-11e8-8bd6-889682651a90.jpg)
![issue4223_screenieb](https://user-images.githubusercontent.com/19371062/40625209-8a2fee6a-62a0-11e8-88ae-45dc3b25e248.jpg)
